### PR TITLE
Added Node.js code samples for calling Netbackup APIs of GET method type

### PR DESCRIPTION
**Pull request by vkiitr**
- Added Node.js modue for loginAPi, getHostDetails and GetNBImages APIs.
- Added sample client code to call above module.
- Added CA certificate support.

### CODE REVIEW

**Comment by vkiitr** [comment]
Reviewers: Please reply back with your thoughts/suggestions on this request.

File: snippets/nodejs/getHostsList.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
+var netbackup = require("./netbackup-module/netbackup.js");
+// including stdio just for creating help
+var stdio = require('stdio');
+var parms = stdio.getopt({
+    'nbmaster': { key: 'n', args: 1, description: 'Master server name', mandatory: true },
+    'port'    : { key: 'pr', args: 1, default: '1556', description: 'Port number' },
+    'username': { key: 'u', args: 1, description: 'User name', mandatory: true },
+    'password': { key: 'p', args: 1, description: 'Password of a user', mandatory: true },
+    'domain'  : { key: 'd', args: 1, default: '', description: 'Domain name (empty by default)' },
```
Comment (shreyas-agnihotri) [inline snippets/nodejs/getHostsList.js]: Do we want to add an option for domain type as well?

File: snippets/nodejs/getNBImages.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+}
+function loginResponse(data) {
+    netbackup.printOnConsole('\nLogin completed!', verbose);
+    jwt = data.token; 
+    netbackup.printOnConsole('\nJWT token : '  + jwt, verbose );
+    
+    netbackup.getNBImages(parms.nbmaster, parms.port, jwt, contentType, verbose, nbImgaes);
+}
+
+function nbImgaes(data) {
```
Comment (shreyas-agnihotri) [inline snippets/nodejs/getNBImages.js]: typo

File: snippets/nodejs/getHostsList.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    'verbose' : { key: 'v', args: 1, description: 'Verbose statements' }
+});
+
+var contentType = "application/vnd.netbackup+json;version=2.0";
```
Comment (kk-github-personal) [inline snippets/nodejs/getHostsList.js]: If the script is for 8.1.1 as well, then the API version should be kept at 1.0. 8.1.1 server won't be able to serve 2.0.

File: README.md [hunk 4]
```diff
@@ -29,6 +29,18 @@ Pre-requisites:
...
+#### Executing the snippets in node.js
+Pre-requisites:
+- NetBackup 8.1.1 or higher
+- node.js v8.9.4 or higher
+- node.js modules: `https, fs, stdio`
+
+Note: To install node.js module, Run `npm install <module_name>`.
+
+Use the following commands to run the node.js samples.
+- `node getHostsList.js --nbmaster <masterServer> -- port <portNumber> --username <username> --password <password> --domain <domainName>`
```
Comment (kk-github-personal) [inline README.md]: I suggest these should be renamed to -nbmaster, username, password, domainName and domainType etc to be consistent with other scripts.

File: snippets/nodejs/netbackup-module/netbackup.js [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+var https = require('https');
+var fs = require('fs');
+
+var exportedMethod = {};
+
+// To disable certificate/ssl verification, uncomment following line
+// process.env.NODE_TLS_REJECT_UNAUTHORIZED = "0";
+
+// TODO save ca certificate at run time and refer that location
+var CA_CERT_PATH
```
Comment (kk-github-personal) [inline snippets/nodejs/netbackup-module/netbackup.js]: Does this mean script can only be run on NetBackup master server?

Comment (vkiitr) [reply snippets/nodejs/netbackup-module/netbackup.js]: I am fixing this by saving ca certificate on local trust store.

File: snippets/nodejs/getNBImages.js (hunk 2 above)
Comment (vkiitr) [reply snippets/nodejs/getNBImages.js]: Done

File: snippets/nodejs/getHostsList.js (hunk 1 above)
Comment (vkiitr) [reply snippets/nodejs/getHostsList.js]: Done

File: snippets/nodejs/getHostsList.js (hunk 3 above)
Comment (vkiitr) [reply snippets/nodejs/getHostsList.js]: Done

File: snippets/nodejs/netbackup-module/netbackup.js (hunk 5 above)
Comment (vkiitr) [reply snippets/nodejs/netbackup-module/netbackup.js]: Fixed this scenario. Now please review the complete code

File: README.md (hunk 4 above)
Comment (vkiitr) [reply README.md]: I tried but with the third party module stdio I am unable to incorporate this. I also tried with other module as well but all are support -- option.

File: snippets/nodejs/netbackup-module/netbackup.js [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+        });
+    } );
+}
+
+function performRequest(option, payload, verbose, returnedRes) {
+    var req = https.request(option, function (response) {
+        printOnConsole('\nStatusCode: ' + response.statusCode, verbose);
+        response.setEncoding('utf-8');
+        var data;
+        response.on('data', function (resBody) {
```
Comment (ghost) [inline snippets/nodejs/netbackup-module/netbackup.js:168]: do you want to print data only in case response status is 200(success)?

Comment (ghost) [reply snippets/nodejs/netbackup-module/netbackup.js:168]: ignore this comment as this code is handling req.on(error)

**Review by ghost — APPROVED** [review]
I dont see any logical error in this code neither any typo. Have worked with vikas and looked at output which is as expected.Hence approving.

**Review by kk-github-personal — APPROVED** [review]

**Pull request merged**